adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -478,10 +481,13 @@ public final void uploadEventMessage(final RoutingContext ctx, final String tena
     private void doUploadMessage(final RoutingContext ctx, final String tenant, final String deviceId,
             final Buffer payload, final String contentType, final Future<MessageSender> senderTracker, final String endpointName) {
 
+        final Integer qosHeader = ctx.request().getHeader(HEADER_QOS_LEVEL) == null ? null : Integer.parseInt(ctx.request().getHeader(HEADER_QOS_LEVEL));
         if (contentType == null) {
             HttpUtils.badRequest(ctx, String.format("%s header is missing", HttpHeaders.CONTENT_TYPE));
         } else if (payload == null || payload.length() == 0) {
             HttpUtils.badRequest(ctx, "missing body");
+        } else if (qosHeader != null && qosHeader != AT_LEAST_ONCE) {
+            HttpUtils.notImplemented(ctx, "only QoS 1 is currently supported");
```

Comment:
this should result in a bad request since it is not a server error.

Reply:
ok, i was thinking there could be support for QoS 2 for telemetry in future. will fix